src/main/webapp/app/js/goal-page.js

```diff
@@ -29,17 +29,17 @@ function onLoad() {
             'Content-Type': 'application/json'
         }
     }).then(function (response) {
-        if (response.status === 200) {
-            return response.json();
-        } else {
-            alert("Something went wrong! error: "+response.status.toString());
-            return false;
-        }
+        displayError(response, 200);
+        return response.json();
```

Comment:
Will this `return` work as intended in case of error?

Reply:
Yes, displayError() does redirect in case of error so the rest of the code will not start.